torch/tensor.py

```diff
@@ -77,6 +78,8 @@ def __deepcopy__(self, memo):
                     new_tensor = self.new()
                     new_tensor.set_(new_storage, self.storage_offset(), self.size(), self.stride())
                     new_tensor.requires_grad = self.requires_grad
+            if self.grad is not None:
+                new_tensor.grad = deepcopy(self.grad)
```

Comment:
I think that for the sharing to work, you need to call this the same way the storage copy is called above: `new_tensor.grad = self.grad.__deepcopy__(memo)`.

If you do that change, don't forget to remove the import for `deepcopy` to make sure the lint is happy.

Reply:
done